Source/RapyutaSimulationPlugins/Private/Sensors/ROS2CameraComponent.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020-2021 Rapyuta Robotics Co., Ltd.
+
+#include "Sensors/ROS2CameraComponent.h"
+
+UROS2CameraComponent::UROS2CameraComponent()
+{
+    // component initialization
+    SceneCaptureComponent = CreateDefaultSubobject<USceneCaptureComponent2D>(TEXT("SceneCaptureComponent"));
+    RenderTarget = CreateDefaultSubobject<UTextureRenderTarget2D>(TEXT("RenderTarget"));
+
+    // Initialize capture and texture component
+    SceneCaptureComponent->SetupAttachment(this);
+}
+
+void UROS2CameraComponent::BeginPlay()
+{
+	Super::BeginPlay();	
+    Init();
+}
+
+void UROS2CameraComponent::Init()
+{
+    SceneCaptureComponent->FOVAngle = FieldOfView;
+    SceneCaptureComponent->OrthoWidth = OrthoWidth;
+    
+    RenderTarget->InitCustomFormat(Width, Height, EPixelFormat::PF_B8G8R8A8, true);
+    SceneCaptureComponent->TextureTarget = RenderTarget;
+
+    // Initialize image data
+    Data.width = Width;
+    Data.height = Height;
+    Data.encoding = Encoding;
+    Data.step = Width * 3; // todo should be variable based on encoding
+    Data.data.AddUninitialized(Width * Height * 3);
+
+    // Node and publisher initialize
+    FActorSpawnParameters SpawnParamsNode;
+    Node = GetWorld()->SpawnActor<AROS2Node>(AROS2Node::StaticClass(), SpawnParamsNode);
+    Node->Name = NodeName.IsEmpty() ? TEXT("UE4CameraNode_" + FGuid::NewGuid().ToString()) : NodeName;

```

Comment:
Just a suggestion, normally I used owner/parent comp's name, which is already made sure to be assigned uniquely by UE, as the affix:
```suggestion
    Node->Name = NodeName.IsEmpty() ? FString::Printf(TEXT("ROS2CameraNode_%s"), *GetName()) : NodeName;
```

Reply:
it will become /ROS2CameraNode_CameraComponent. I guess component name is not unique.

Reply:
Each, all actors/components in UE are basically `UObject`, of which the names must be unique, as noted [here](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/CoreUObject/Public/UObject/UObjectGlobals.h#L1163), or as in [UObject::Rename()](https://docs.unrealengine.com/4.27/en-US/API/Runtime/CoreUObject/UObject/UObject/Rename/)
-> So `GetName()` returns that object name, versus `GetClassName()` which returns the type name as `CameraComponent`.

But yeah, if we then might run multiple Sim instances, `GUID` would be more reliable I suppose.

Reply:
All actors/components in UE are basically `UObject`, of which the names must be unique, as noted [here](https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/CoreUObject/Public/UObject/UObjectGlobals.h#L1163), or as in [UObject::Rename()](https://docs.unrealengine.com/4.27/en-US/API/Runtime/CoreUObject/UObject/UObject/Rename/)
-> So `GetName()` returns that object name, versus `GetClassName()` which returns the type name as `CameraComponent`.

But yeah, if we then might run multiple Sim instances, `GUID` would be more reliable I suppose.